src/include/storage/index/b_plus_tree_debug.h

```diff
@@ -0,0 +1,275 @@
+//===----------------------------------------------------------------------===//
+//
+//                         BusTub
+//
+// b_plus_tree_debug.h
+//
+// Identification: bustub/src/include/storage/index/b_plus_tree_debug.h
+//
+// Copyright (c) 2024, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include <sstream>
+#include <string>
+
+#include "common/exception.h"
+#include "common/logger.h"
+#include "common/rid.h"
+#include "storage/index/b_plus_tree.h"
+#include "storage/page/b_plus_tree_page.h"
+
+namespace bustub {
+
+/*****************************************************************************
+ * UTILITIES AND DEBUG
+ *****************************************************************************/
+
+/*
+ * This method is used for test only
+ * Read data from file and insert one by one
+ */
+INDEX_TEMPLATE_ARGUMENTS
+void BPLUSTREE_TYPE::InsertFromFile(const std::filesystem::path &file_name) {
+  int64_t key;
+  std::ifstream input(file_name);
+  while (input) {
+    input >> key;
+
+    KeyType index_key;
+    index_key.SetFromInteger(key);
+    RID rid(key);
+    Insert(index_key, rid);
+  }
+}
+/*
+ * This method is used for test only
+ * Read data from file and remove one by one
+ */
+INDEX_TEMPLATE_ARGUMENTS
+void BPLUSTREE_TYPE::RemoveFromFile(const std::filesystem::path &file_name) {
+  int64_t key;
+  std::ifstream input(file_name);
+  while (input) {
+    input >> key;
+    KeyType index_key;
+    index_key.SetFromInteger(key);
+    Remove(index_key);
+  }
+}
+
+INDEX_TEMPLATE_ARGUMENTS
+void BPLUSTREE_TYPE::Print(BufferPoolManager *bpm) {
+  auto root_page_id = GetRootPageId();
+  if (root_page_id != INVALID_PAGE_ID) {
+    auto guard = bpm->ReadPage(root_page_id);
+    PrintTree(guard.GetPageId(), guard.template As<BPlusTreePage>());
+  }
+}
+
+INDEX_TEMPLATE_ARGUMENTS
+void BPLUSTREE_TYPE::PrintTree(page_id_t page_id, const BPlusTreePage *page) {
+  if (page->IsLeafPage()) {
+    auto *leaf = reinterpret_cast<const LeafPage *>(page);
+    std::cout << "Leaf Page: " << page_id << "\tNext: " << leaf->GetNextPageId() << std::endl;
+
+    // Print the contents of the leaf page.
+    std::cout << "Contents: ";
+    for (int i = 0; i < leaf->GetSize(); i++) {
+      std::cout << leaf->KeyAt(i);
+      if ((i + 1) < leaf->GetSize()) {
+        std::cout << ", ";
+      }
+    }
+    std::cout << std::endl;
+    std::cout << std::endl;
+
+  } else {
+    auto *internal = reinterpret_cast<const InternalPage *>(page);
+    std::cout << "Internal Page: " << page_id << std::endl;
+
+    // Print the contents of the internal page.
+    std::cout << "Contents: ";
+    for (int i = 0; i < internal->GetSize(); i++) {
+      std::cout << internal->KeyAt(i) << ": " << internal->ValueAt(i);
```

Comment:
`internal->KeyAt(0)` is an undefined value right? maybe don't print this out or state that it's irrelevant in the output.